services/viva-ms/lambdas/submitApplication.js

```diff
@@ -3,86 +3,132 @@ import AWS from 'aws-sdk';
-  const {
-    forms: {
-      [caseItem.currentFormId]: { answers },
-    },
-    details: { workflowId },
-    pdf: pdfBinaryBuffer,
-  } = caseItem;
+  const { PK, SK, pdf: pdfBinaryBuffer } = caseItem;
+  const personalNumber = PK.substring(5);
 
-  const [applicationPostError, applicationResponse] = await to(
+  const [vivaPostError, vivaApplicationResponse] = await to(
     vivaAdapter.application.post({
       applicationType: 'recurrent',
       personalNumber,
-      workflowId,
-      answers,
+      workflowId: caseItem.details?.workflowId || '',
+      answers: caseItem.forms[recurringFormId].answers,
       rawData: pdfBinaryBuffer.toString(),
       rawDataType: 'pdf',
     })
   );
-  if (applicationPostError) {
+  if (vivaPostError) {
     log.error(
-      'Application post error',
+      'Viva adapter submit application failed',
       context.awsRequestId,
-      'service-viva-ms-submitApplication-002',
-      applicationPostError
+      'service-viva-ms-submitApplication-001',
+      {
+        axios: { ...vivaPostError },
+      }
     );
+    return false;
+  }
 
-    return;
+  if (notApplicationReceived(vivaApplicationResponse)) {
+    log.error(
+      'Viva application receive failed',
+      context.awsRequestId,
+      'service-viva-ms-submitApplication-002',
+      { vivaResponse: { ...vivaApplicationResponse } }
+    );
+    return false;
   }
 
-  log.info(
-    'Application response',
-    context.awsRequestId,
-    'service-viva-ms-submitApplication-003',
-    applicationResponse
-  );
+  if (!vivaApplicationResponse?.id) {
+    log.error(
+      'Viva application response does not contain workflow id',
+      context.awsRequestId,
+      'service-viva-ms-submitApplication-003',
+      vivaApplicationResponse
+    );
+    return false;
+  }
 
-  const caseKeys = {
-    PK: caseItem.PK,
-    SK: caseItem.SK,
+  const caseKeys = { PK, SK };
+  const newCaseValues = {
+    state: VIVA_APPLICATION_RECEIVED,
+    workflowId: vivaApplicationResponse.id,
   };
+  const [updateError, newVivaCase] = await to(updateVivaCase(caseKeys, newCaseValues));
+  if (updateError) {
+    log.error(
+      'Database update viva case failed',
+      context.awsRequestId,
+      'service-viva-ms-submitApplication-004',
+      updateError
+    );
+    return false;
+  }
 
-  const [putEventError] = await to(
-    putEvent(
-      {
-        caseKeys,
-      },
-      'vivaMsSubmitApplicationSuccess',
-      'vivaMs.submitApplication'
-    )
-  );
+  const [putEventError] = await to(putVivaMsEvent.applicationReceivedSuccess({ caseKeys }));
   if (putEventError) {
-    throw putEventError;
+    log.error(
+      'Put event: applicationReceivedSuccess failed',
+      context.awsRequestId,
+      'service-viva-ms-submitApplication-005',
+      putEventError
+    );
+    return false;
   }
+
+  log.info('Updated viva case successfully', context.awsRequestId, null, newVivaCase);
+  return true;
+}
+
+function notApplicationReceived(response) {
```

Comment:
`notApplicationReceived`, `noApplicationReceived` or just `applicationReceived` ?

Perhaps: (to avoid using the word "not" in function name, which makes you think alot more when reading the code)

```js
function applicationReceived() {
  return response?.status === 'OK'
}
```
And use it like:

```js
if (!applicationReceived(response)) {
 ...
}
```

Reply:
I like the pattern of returning truthy values from functions. However i find it easy to miss the Not `!` operator i js when reading through code. 

Could this be a compromise perhaps?
```JS

applicationNotReceived = !applicationRecived()

if(applicationNotRecived) {
 ...
}

```

Reply:
Will be fixed in future task